R/helpers.R

```diff
@@ -10,41 +10,58 @@
 #' @param valuecol, name of the column containing the GDP values, default is "weight".
 #' @param usecache, store the result in a RDS file in the working directory, default is FALSE.
 #' @param gdpfile, if caching is required, specify the filename here, default is "GDPcache.rds"
+#' @param isolev, if the data provided has to be on ISO or regional level
 #' @keywords gdp
 #' @import data.table
 #' @importFrom madrat calcOutput
 #' @export
 
-getRMNDGDP <- function(scenario="gdp_SSP2",
-                      yearcol="year",
-                      isocol="iso",
-                      valuecol="weight",
-                      usecache=F,
-                      gdpfile="GDPcache.rds"){
+getRMNDGDPcap <- function(scenario="gdp_SSP2",
+                          yearcol="year",
+                          isocol,
+                          valuecol="weight",
+                          isolev = F,
+                          usecache = F,
+                          gdpfile ="GDPcache.rds"){
 
-    variable <- Year <- NULL
+  `.` <- iso <- value <- GDP_cap <- weight <- POP_val <- region <- variable <- Year <- NULL
 
-    if(usecache && file.exists(gdpfile)){
```

Comment:
maybe just use the original code here, and let the user specify the cache filename.